app/models/concerns/spree/adyen/payment.rb

```diff
@@ -125,30 +129,20 @@ def process &block
         end
       end
 
-      def rest_client
-        ::Adyen::REST::Client.new(
-          ::Adyen.configuration.environment,
-          payment_method.api_username,
-          payment_method.api_password
-        )
-      end
-
       def authorize_new_payment
-        ::Adyen::REST.session(rest_client) do |client|
-          # If this is a new credit card we should have the encrypted data
-          if source.encrypted_data
-            client.authorise_recurring_payment(payment_params.merge(encrypted_card_data))
-          elsif source.gateway_customer_profile_id
-            client.reauthorise_recurring_payment(payment_params)
-          else
-            raise Spree::Core::GatewayError.new(
-              I18n.t(:missing_encrypted_data, scope: 'solidus-adyen')
-            )
-          end
+        # If this is a new credit card we should have the encrypted data
+        if source.encrypted_data
```

Comment:
would be nice if we had predicate defined on the source for this. The fact that the source has encrypted data does not imply to me, as an outsider, that this is a new credit card.

Something like `source.new_credit_card?` would be nice.